src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/params.py

```diff
@@ -0,0 +1,47 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+from ._util import ParametersContext
+
+with ParametersContext(command='monitor alert-rules') as c:
+    c.register_alias('name', ('--azure-resource-name',))
+    c.register_alias('rule_name', ('--name', '-n'))
+
+with ParametersContext(command='monitor alert-rule-incidents') as c:
+    c.register_alias('incident_name', ('--name', '-n'))
+
+with ParametersContext(command='monitor log-profiles') as c:
+    c.register_alias('log_profile_name', ('--name', '-n'))
+
+with ParametersContext(command='monitor autoscale-settings') as c:
+    c.register_alias('name', ('--azure-resource-name',))
+    c.register_alias('autoscale_setting_name', ('--name', '-n'))
+
+with ParametersContext(command='monitor alert-rules create') as c:
+    from azure.cli.command_modules.monitor.sdk.models.alert_rule_resource import AlertRuleResource
+
+    c.expand('parameters', AlertRuleResource)
+    c.register_type('condition', ('--condition',),
+                    json.loads,
+                    'JSON encoded condition configuration. Use @{file} to load from a file.')
```

Comment:
I recommend you spell out the kwargs like we do in the traditional model. (i.e. type=json.loads, help='blahblah')